Add email post type

merkushin opened the pull request:
Fixes #6458 

### Changes proposed in this Pull Request

* Add the feature flag: `sensei_feature_flag_email_customization`.
* Add the Email post type: `sensei_email`.

### Testing instructions

* Enable the feature flag: `add_filter( 'sensei_feature_flag_email_customization', '__return_true' );`
* Go to http://{your-dev-wordpress-site}/wp-admin/edit.php?post_type=sensei_email

### Screenshot / Video
<img width="1680" alt="CleanShot 2023-02-06 at 14 58 17@2x" src="https://user-images.githubusercontent.com/329356/216916183-bd866f47-4025-45bb-ae18-27bb4dbc6614.png">

merkushin committed:
Add email post type

merkushin committed:
Fix namespace in test

gabrielcaires commented on includes/class-sensei.php:
```diff
@@ -568,6 +569,11 @@ public function initialize_global_objects() {
 		// Cron for periodically cleaning guest user related data.
 		Sensei_Temporary_User_Cleaner::instance()->init();
+
+		$email_customization_enabled = $this->feature_flags->is_enabled( 'email_customization' );
```
Can we have a test to cover this checking, it is simple, but it helps us ensure we are not leaking features that are not ready yet.

merkushin replied:
Tests added here: https://github.com/Automattic/sensei/pull/6487/commits/40ef52a98e5afa7eb7725c6c965a02454b426fe8

gabrielcaires replied:
Oh, I was trying to say, the test should cover if the feature flag is off, the post type should not be registered.

gabrielcaires commented on includes/class-sensei-autoloader.php:
```diff
@@ -285,6 +285,11 @@ public function initialize_class_file_map() {
+			/**
+			 * Email Customization
+			 */
+			\Sensei\Internal\Emails\Email_Post_Type::class => 'internal/emails/class-email-post-type.php',
```
Is possible only load this one when the feature is on, and then make the email_post type auto-initializable?

merkushin replied:
This code makes the class available for usage (so you don't need to `require` the file before using the class, it is a PHP-specific thing). It does't instantiate it.
I think it is better not to wrap in the feature flag.

gabrielcaires reviewed (dismissed):
I tested and all works well with the feature on and off, I left some comments just to align the strategy but I think they are not blockers because the custom post type is hidden even with the feature ON (accessible only via URL).

merkushin committed:
Add test for email customization flag feature

merkushin committed:
Fix linter issues: remove whitespaces

gabrielcaires reviewed (changes requested):
~Reviewing the card scope, I just noticed we are not adding the taxonomy to separate students emails and teachers' emails. 
Are you planning to convert in a different PR~

Move the scope to another card.

gabrielcaires committed:
Testing install only the chromium dependencies

The pipeline is falling due to network connection issues installing the playwright dependencies.
